packages/ag-charts-community/src/util/properties.ts

```diff
@@ -84,6 +84,38 @@ export class PropertiesArray<T extends BaseProperties> extends Array {
     }
 }
 
+export class TypedPropertiesArray<T extends BaseProperties> extends PropertiesArray<T> {
```

Comment:
@iMoses could you review this part?

Reply:
It doesn't seem to be used. Am I missing something?

Reply:
It is used here https://github.com/ag-grid/ag-charts/pull/1538/files#diff-fde1cb072dc3708f277225348f863116c84ddf03dd09aad886b49be27fa0a65cR84 with this definition https://github.com/ag-grid/ag-charts/pull/1538/files#diff-fde1cb072dc3708f277225348f863116c84ddf03dd09aad886b49be27fa0a65cR33-R37